server/cpho/views/upload_indicators.py

```diff
@@ -0,0 +1,437 @@
+        ),
+        "Province_NS": DimensionValue.objects.get(
+            dimension_type__code="province", value="ns"
+        ),
+        "Province_NL": DimensionValue.objects.get(
+            dimension_type__code="province", value="nl"
+        ),
+        "Province_PE": DimensionValue.objects.get(
+            dimension_type__code="province", value="pe"
+        ),
+        "Province_NU": DimensionValue.objects.get(
+            dimension_type__code="province", value="nu"
+        ),
+        "Province_NT": DimensionValue.objects.get(
+            dimension_type__code="province", value="nt"
+        ),
+        "Province_YT": DimensionValue.objects.get(
+            dimension_type__code="province", value="yt"
+        ),
+        "Region_ATLANTIC": DimensionValue.objects.get(
+            dimension_type__code="region", value="atlantic"
+        ),
+        "Region_PRAIRIE": DimensionValue.objects.get(
+            dimension_type__code="region", value="prairies"
+        ),
+        "Region_TERRITORIES": DimensionValue.objects.get(
+            dimension_type__code="region", value="territories"
+        ),
+        "Gender_MEN": DimensionValue.objects.get(
+            dimension_type__code="gender", value="men"
+        ),
+        "Gender_WOMEN": DimensionValue.objects.get(
+            dimension_type__code="gender", value="women"
+        ),
+        "Gender_BOYS": DimensionValue.objects.get(
+            dimension_type__code="gender", value="boys"
+        ),
+        "Gender_GIRLS": DimensionValue.objects.get(
+            dimension_type__code="gender", value="girls"
+        ),
+        "Sex_MALES": DimensionValue.objects.get(
+            dimension_type__code="sex", value="m"
+        ),
+        "Sex_FEMALES": DimensionValue.objects.get(
+            dimension_type__code="sex", value="f"
+        ),
+        "Canada_CANADA": DimensionValue.objects.get(
+            dimension_type__code="canada", value="canada"
+        ),
+    }
+
+    def save(self):
+        data = self.cleaned_data["csv_file"]
+        # TODO: Update Periods
+        preiod_val = Period.objects.get(year="2023")
+        for datum in data:
+            indicator_obj, ind_created = Indicator.objects.get_or_create(
+                name=datum["Indicator"],
+                detailed_indicator=datum["Detailed Indicator"],
+                sub_indicator_measurement=datum["Sub_Indicator_Measurement"],
+                category=category_mapper[datum["Category"]],
+                sub_category=subcategory_mapper[datum["Topic"]],
+            )
+
+            dim_val = None
+            lit_dim_val = None
+            if datum["Dimension_Type"] != "Age Group":
+                dim_val = non_literal_dimension_value_mapper[
+                    f'{datum["Dimension_Type"]}_{datum["Dimension_Value"]}'
+                ]
+            else:
+                lit_dim_val = datum["Dimension_Value"]
+
+            indData_obj, id_created = IndicatorDatum.objects.get_or_create(
```

Comment:
get_or_create will always create a new record whenever there's even a single new value. What if they want to update an existing value?

We can use the (indicator, dimension_type, dimension_value, literal_dimension_val) to identify an existing row. If it exists, write to it, otherwise create a new record.